### PR TITLE
fix(clipboard): 

### PR DESCRIPTION
**Pull request by aditiohri**
## Pull request description
* Adds a check for window.isSecureContext signifying a secure HTTPS connection ensuring navigator.clipboard API will work correctly. Otherwise, throws an error.
* Hides element using hidden attribute if an error is thrown as the element would be unusable in that state possibly confusing the end-user.
* Hides element using the aria-disabled attribute if the element has been hidden.

## tldr; 
If the site is not secure, and the clipboard fallback cannot be used,  an error is thrown and the element is hidden for all users.

### Related issues

#1868 - [pfe-clipboard] If site isn't HTTPS copy functionality will not work.
#1873 -  bug: pfe-clipboard does not have an accessible experience 

### Preview

1. Check out to the branch fix/CPCORE-6611-pfeClipboard-httpsOnly 
2. Run npm start
3. Go to https://deploy-preview-1981--patternfly-elements.netlify.app/components/clipboard/ or http://localhost:8000/demo/pfe-clipboard on your local

### What has changed and why

- pfe-clipboard.ts
    * add a private state variable, `_secure`, to check for [`window.isSecureContext` value](https://developer.mozilla.org/en-US/docs/Web/API/isSecureContext)
   * add a `_setCopyType` function that sets `copyType` property depending on the value of `_secure`
   * uses `copyType` in `copyTextToClipboard`

### Testing instructions

- In a secure context
  1. Set up the branch locally
  2. Navigate to the pfe-clipboard
  3. User should be able to see and click on the element 
  4. Screen reader should allow you to interact with the element
- In an insecure context
  1. Set up the branch locally
  2. On line 111, set `_secure` to `false`
  3. Navigate to the pfe-clipboard
  4. User should not be able to see or click on the element 
  5. Screen reader should not allow you to interact with the element

#### Browser requirements

Your component should work in all of the following environments:

- [x] Latest 2 versions of Edge
- [x] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [x] Firefox 78 (or latest version for Red Hat Enterprise Linux distribution)
- [x] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [x] Latest 2 versions of Safari
- [x] Android mobile device (such as the Galaxy S9)
- [x] Apple mobile device (such as the iPhone X)
- [x] Apple tablet device (such as the iPhone Pro)

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [ ] Tests have been updated to cover these changes.
- [x] Browser testing passed.
- [x] Changelog updated.
- [ ] Documentation (README.md, WHY.md, etc.) updated or added.
- [ ] Link to the demo recording: []()
- [ ] Approved by designer.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

**Comment by wesruv** [comment]
Logging Gophers Take Mopeds! (looks good to me)

**Review by wesruv — APPROVED** [review]

File: elements/pfe-clipboard/pfe-clipboard.ts [hunk 1]
```diff
@@ -104,6 +105,17 @@ export class PfeClipboard extends LitElement {
...
 
+  /**
+   * Specify if current site is behind [https]
+   */
+  @state() private _secure = window.isSecureContext;
+
+  /**
+   * Specify the copy type to be used depending on security and browser support
+   */
+  @property({ type: String || null }) copyType: 'navigator'|'queryCommand'| null = null;
```
Comment (bennypowers) [inline elements/pfe-clipboard/pfe-clipboard.ts:116]:
```suggestion
  @property({ type: String }) copyType: 'navigator' | 'queryCommand'  | null = null;
```

we don't need the `null` in the property declaration. `type` here is a shorthand for `converter`. see https://lit.dev/docs/components/properties/#property-options

File: elements/pfe-clipboard/pfe-clipboard.ts [hunk 2]
```diff
@@ -76,6 +76,7 @@ export class PfeClipboard extends LitElement {
 
   static readonly styles = [style];
 
+
```
Comment (bennypowers) [inline elements/pfe-clipboard/pfe-clipboard.ts:79]:
```suggestion
```

**Review by bennypowers — CHANGES_REQUESTED** [review]
Two small changes please, otherwise LGTM

**Review by heyMP — APPROVED** [review]
Let's Go Thursday Managers 👍

**Review by heyMP — APPROVED** [review]

**Pull request merged**